<a href="https://colab.research.google.com/github/wolfram-roemhild/deep-learning-colab/blob/master/TensorFlow_MNIST.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##### Copyright 2018 The TensorFlow Authors.

Licensed under the Apache License, Version 2.0 (the "License");

In [0]:
#@title Licensed under the Apache License, Version 2.0 (the "License"); { display-mode: "form" }
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Get Started with TensorFlow - MNIST short example

#### Google Colaboratory

This is a [Google Colaboratory](https://colab.research.google.com/notebooks/welcome.ipynb) notebook file. Python programs are run directly in the browser—a great way to learn and use TensorFlow. To run the Colab notebook:

1. Connect to a Python runtime: At the top-right of the menu bar, select *CONNECT*.
2. Run all the notebook code cells: Select *Runtime* > *Run all*.

For more examples and guides (including details for this program), see [Get Started with TensorFlow](https://www.tensorflow.org/get_started/).

Let's get started, import the TensorFlow library into your program:

#### TensorFlow Model

In [0]:
import tensorflow as tf

Load and prepare the [MNIST](http://yann.lecun.com/exdb/mnist/) dataset. Convert the samples from integers to floating-point numbers:

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11501568/11490434 [==============================] - 0s 0us/step


Build the `tf.keras` model by stacking layers. Select an optimizer and loss function used for training:

In [0]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(512, activation=tf.nn.relu),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate model (only 7 epochs):

In [5]:
model.fit(x_train, y_train, epochs=7)

Epoch 1/7
60000/60000 [==============================] - 15s 255us/step - loss: 0.1986 - acc: 0.9411
Epoch 2/7
60000/60000 [==============================] - 15s 244us/step - loss: 0.0801 - acc: 0.9756
Epoch 3/7
60000/60000 [==============================] - 15s 247us/step - loss: 0.0543 - acc: 0.9829
Epoch 4/7
60000/60000 [==============================] - 15s 246us/step - loss: 0.0362 - acc: 0.9884
Epoch 5/7
60000/60000 [==============================] - 14s 240us/step - loss: 0.0272 - acc: 0.9909
Epoch 6/7
60000/60000 [==============================] - 15s 243us/step - loss: 0.0200 - acc: 0.9936
Epoch 7/7
60000/60000 [==============================] - 14s 240us/step - loss: 0.0186 - acc: 0.9939


In [6]:
model.evaluate(x_test, y_test)

10000/10000 [==============================] - 1s 56us/step


[0.06685141461325111, 0.9801]

You’ve now trained an image classifier with ~98% accuracy on this dataset. See [Get Started with TensorFlow](https://www.tensorflow.org/get_started/) to learn more.